### DISCUSSION TITLE
Our actions are failing for external pull requests but work fine when running locally

### DISCUSSION THREAD

**Discussion by dabeeeenster** [community/community#26764] [Actions]
<p>We’re seeing a really odd issue, where Github actions results in a different outcome depending on whether the very same code is being run from an external PR as opposed to an internal one.</p>
<p>For example, this PR: <a href="https://github.com/Flagsmith/flagsmith/pull/449#partial-pull-merging" class="inline-onebox" rel="noopener nofollow ugc">Allow sentry-trace header in CORS restrictions by tm1000 · Pull Request #449 · Flagsmith/flagsmith · GitHub</a> fails our end to end tests, but the same code that I made a PR from directly within the repo <a href="https://github.com/Flagsmith/flagsmith/pull/458" class="inline-onebox" rel="noopener nofollow ugc">improvement/sentry cors by dabeeeenster · Pull Request #458 · Flagsmith/flagsmith · GitHub</a> is passing.</p>
<p>I can’t explain this - is there something I’m missing RE differences between these action environments?</p>

**Comment by korthout** [comment]
<p>I don’t immediately see what’s wrong here, but the <code>pull_request</code> event does not run in the same context on external PRs. You should probably use <code>pull_request_target</code> (<a href="https://docs.github.com/en/actions/learn-github-actions/events-that-trigger-workflows#pull_request_target" rel="noopener nofollow ugc">docs</a>) for pull requests from forked repos. Be aware of the security warning provided with those docs.</p>
<p>Hope it helps.</p>

**Comment by jku** [comment, marked as answer]
<p>A potential difference is secrets: runs for PRs coming from forks do not get access to secrets (other than GITHUB_TOKEN), and your workflow seems to be using one…</p>

**Comment by dabeeeenster** [comment]
<p>Thanks for the answers here - just to say if anyone finds this thread - it was due to these external PRs not having access to our repo secrets. Thanks for the replies!</p>
